mlflow/store/artifact/utils/models.py

```diff
@@ -10,8 +10,10 @@
 
 # This regex is used by _parse_model_uri and details for the regex match
 # can be found in _improper_model_uri_msg.
+_DISALLOWED_CHARACTERS = r"\\\/$@"
+
 _MODEL_URI_REGEX = re.compile(
-    r"^\/(?P<model_name>[\w \.\-]+)(\/(?P<suffix>[\w]+))?(@(?P<alias>[\w\-]+))?$"
+    rf"^\/(?P<model_name>[^{_DISALLOWED_CHARACTERS}]+)(\/(?P<suffix>[\w]+))?(@(?P<alias>[\w\-]+))?$"
```

Comment:
```suggestion
    rf"^\/(?P<model_name>[^\\\/$@]+)(\/(?P<suffix>[\w]+))?(@(?P<alias>[\w\-]+))?$"
```

Reply:
lol good point. No need to externalize that short list. I'll change it :)